### ISSUE TITLE
Shouldn't the default file destination be the Cache directory?

### ISSUE THREAD

**Issue by xeo-it** [SwiftyBeaver/SwiftyBeaver#68]
Currently the FileDestination class save a log file in the Documents directory of the app.
Since there is no control for the file size (this could be a different issue actually), the file could grow indefinitely.  
Considering that the documents directory is backed up to iCloud, the cache directory in my opinion would be a better location.

**Comment by skreutzberger** [comment]
that’s a good point! How does iOS deal with the cache directory in detail? Is it cleared after a while or just not synced to iCloud?

**Comment by xeo-it** [comment]
From [apple docs](https://developer.apple.com/library/ios/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1) 
> `Use this directory to write any app-specific support files that your app can re-create easily. Your app is generally responsible for managing the contents of this directory and for adding and deleting files as needed.
In iOS 2.2 and later, the contents of this directory are not backed up by iTunes. In addition, iTunes removes files in this directory during a full restoration of the device.
On iOS 5.0 and later, the system may delete the Caches directory on rare occasions when the system is very low on disk space. This will never occur while an app is running. However, you should be aware that iTunes restore is not necessarily the only condition under which the Caches directory can be erased.

Basically it's not saved on iCloud and can be deleted if the device is running low on disk space.
Usually logs file are stored here.

**Comment by skreutzberger** [comment]
:+1: can you do a PR?

**Comment by prenagha** [comment]
Or isn't there a Logs directory? At least on Mac there usually is. Not sure if that is standard across platforms or not.
I would be careful with Cache, as the apple docs note it is for `files that your app can re-create easily`. Log files can not be recreated, once a log file is deleted its content cannot be re-created.

**Comment by xeo-it** [comment]
There is no Logs directory in IOS, maybe I can use the Cache directory for iOS and leave Documents for the Mac target.  
Usually the Cache directory is cleaned when you restore the phone or in very few rare cases, this should not concern too much, and on a iOS app usually the logs are re-created easily since the apps follow a standard life cycle.  

On a side note, it would be nice to add few features to the file destination:
- Persistency: the log can be deleted and re-created at every app start (just for iOS), this is because most of the time people want to see just the latest log, and if an error occur usually they send the logs via crash report and then delete the file
- File deletion: Ex: I downloaded the log, I don't need it anymore on the device
- Daily logs: if you have persistent logs active, it's not very nice to skim a 20mb text file, having one file per day makes this easy
- Retention days: if persistency is on, the files older than this value are deleted

This are only few of options that I usually use in my projects.  
Should I create a new issue to discuss these maybe?

**Comment by skreutzberger** [comment]
The change was implemented by e1b21238fff8eb - including a new OS constant which the framework can use to better know what platform/OS it is running on.
